### PR TITLE
#34 follow unfollow

### LINKED ISSUE #34: Implement follow/unfollow feature (backend)
User story #33

**Description**
Implement the relationship between users in the database to show who follows who.

Notes:
-Modify user database for the followers/following
-POST request for follow/unfollow
-Integrate with the front-end

**Has to be done first**

### PR DESCRIPTION
**Pull request by silentswordxv**
Feature has been implemented.

closes #34

### CODE REVIEW

**Comment by silentswordxv** [comment]
TODO: After merge or before, add test script.

**Review by justpino7 — APPROVED** [review]
Looks good to me!

**Review by guaj — APPROVED** [review]
Tested, on user profile, works well

File: client/src/Components/Templates/UserProfile/UserProfile.js [hunk 1]
```diff
@@ -133,12 +126,37 @@ export default function UserProfile(props) {
...
     const classes = useStyles();
 
-        // Checking the backend to see if the user is logged in
+    // Checking the backend to see if the user is logged in
     useEffect(() => {
         if(localStorage.getItem("user") === null)
             window.location = "/login#redirect";
     }, [])
 
+    const usernameS = {currentUsername: JSON.parse(localStorage.getItem("user")).username, visitedUsername: window.location.href.split("/")[4]};
```
Comment (I3alan) [inline client/src/Components/Templates/UserProfile/UserProfile.js:135]: it's not a deal breaker, but i think we should name the variable a bit more descriptively.

**Review by jfv-2000 — APPROVED** [review]
👍 Looks good to me

**Pull request merged**